### PR TITLE
Optimization: Use mcopy to copy array in add liquidity unbalanced

### PR DESCRIPTION
**Pull request by jubeira**
# Description

Super minor optimization: use `mcopy` instead of a naive loop.
I don't see a good way to test this though. Everything works, but e.g. if I copy more memory on purpose there's no failure anywhere. I'd only proceed with an auditor's blessing.

Update: added extra paranoid tests; they should fail if the boundaries of `mcopy` are incorrect. This should be ok.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [x] Optimization: [x] gas / [x] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- N/A Complex code has been commented, including external interfaces
- N/A Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

**Review by danielmkm — APPROVED** [review]
Would appreciate auditor input, but the change itself looks simple and correct

**Comment by jubeira** [comment]
> Would appreciate auditor input, but the change itself looks simple and correct

Cantina guys already checked it out; it's just that I delayed the merge.

File: pkg/solidity-utils/contracts/helpers/ScalingHelpers.sol [hunk 1]
```diff
@@ -111,8 +111,8 @@ library ScalingHelpers {
         uint256 length = from.length;
         InputHelpers.ensureInputLengthMatch(length, to.length);
 
-        for (uint256 i = 0; i < length; ++i) {
```
Comment (elshan-eth) [inline pkg/solidity-utils/contracts/helpers/ScalingHelpers.sol:114]: Nice savings!

**Review by elshan-eth — APPROVED** [review]
LGTM!
